mava/wrappers/matrax.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+import chex
+import jax.numpy as jnp
+from jumanji import specs
+from jumanji.env import Environment
+from jumanji.types import TimeStep
+from jumanji.wrappers import Wrapper
+
+from mava.types import Observation, State
+
+
+class MatraxWrapper(Wrapper):
+    def __init__(self, env: Environment):
+        super().__init__(env)
+        self._num_agents = self._env.num_agents
+        self.time_limit = self._env.time_limit
+        self._num_actions = self._env.num_actions
+        self.num_obs_features = self._env.observation_spec().agent_obs.shape[-1]
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for `step` and `reset`."""
+        observation = Observation(
+            agents_view=timestep.observation.agent_obs,
+            action_mask=jnp.ones((self._num_agents, self._num_actions), dtype=bool),
```

Comment:
Since this is fixed, can we move it up so we don't have to calculate it over and over at each timestep:
```suggestion
            action_mask=self.action_mask

```